test/e2e/util.go

```diff
@@ -576,3 +576,30 @@ func listPods(c *client.Client, namespace string, label labels.Selector, field f
 	}
 	return pods, err
 }
+
+// Verifies whether there were no than <limit> bad events and prints their histogram.
+func VerifyBadEvents(events []*api.Event, limit int) {
+	type histogramKey struct {
+		reason string
+		source string
+	}
+	histogram := make(map[histogramKey]int)
+	for _, e := range events {
+		histogram[histogramKey{reason: e.Reason, source: e.Source.Component}]++
+	}
+	for key, number := range histogram {
+		Logf("- reason: %s, source: %s -> %d", key.reason, key.source, number)
+	}
+	badPatterns := []string{"kill", "fail"}
+	badEvents := 0
+	for key, number := range histogram {
+		for _, s := range badPatterns {
+			if strings.Contains(key.reason, s) {
+				Logf("WARNING %d events from %s with reason: %s", number, key.source, key.reason)
+				badEvents += number
+				break
+			}
+		}
+	}
+	Expect(badEvents).NotTo(BeNumerically(">", limit))
```

Comment:
It would be better to have this return something and let the caller fail on the return status.  Maybe this could just be a function that returns the total count of bad events or the event histogram?  If you are going to do Expects in a function then you need to add "defer GinkgoRecover()" in the function so the exception is properly caught.  This is usually done at the top.